docs/rhel4edge_iso.md

```diff
@@ -0,0 +1,212 @@
+
+## Build RHEL for Edge Installer ISO
+Log into the development virtual machine with the `microshift` user credentials.
+> The development machine configuration guidelines can be found in the [MicroShift Development Environment on RHEL 8.x](./devenv_rhel8.md) document.
+
+Follow the instructions in the [RPM Packages](./devenv_rhel8.md#rpm-packages) section to create MicroShift RPM packages.
+
+The scripts for building the installer are located in the `scripts/image-builder` subdirectory.
+
+### Prerequisites
+Execute the `scripts/image-builder/configure.sh` script to install the tools necessary for building the installer image.
+
+Download the OpenShift pull secret from the https://console.redhat.com/openshift/downloads#tool-pull-secret page and save it into the `~microshift/pull-secret.txt` file. 
+
+Make sure there is more than 20GB of free disk space necessary for the build artifacts. Run the following command to free the space if necessary.
+```bash
+./scripts/image-builder/cleanup.sh -full
+```
+
+Note that the command deletes various user and system data, including:
+- The `scripts/image-builder/_builds` directory containing image build artifacts
+- MicroShift `ostree` server container and all the unused container images
+- All the Image Builder jobs are canceled and deleted
+- Project-specific Image Builder sources are deleted
+- The user `~/.cache` and `/tmp/containers` directory contents are deleted to clean various cache files
+- The `/var/cache/osbuild-worker` directory contents are deleted to clean Image Builder cache files
+
+### Building Installer
+It is recommended to execute the partial cleanup procedure (without `-full` argument) before each build to reclaim cached disk space from previous builds. 
+```bash
+./scripts/image-builder/cleanup.sh
+```
+Note that the command deletes various user and system data, including:
+- MicroShift `ostree` server container
+- All the Image Builder jobs are canceled
+- Project-specific Image Builder sources are deleted
+- The `/var/cache/osbuild-worker` directory contents are deleted to clean Image Builder cache files
+
+Run the build script without arguments to see its usage.
+```bash
+./scripts/image-builder/build.sh
+Usage: build.sh <-pull_secret_file path_to_file> [-ostree_server_name name_or_ip] [-custom_rpms /path/to/file1.rpm,...,/path/to/fileN.rpm]
+   -pull_secret_file   Path to a file containing the OpenShift pull secret
+   -ostree_server_name Name or IP address of the OS tree server (default: )
+   -custom_rpms        Path to one or more comma-separated RPM packages to be included in the image
+
+Note: The OpenShift pull secret can be downloaded from https://console.redhat.com/openshift/downloads#tool-pull-secret.
+```
+
+Continue by running the build script with the pull secret file argument and wait until build process is finished. It may take over 30 minutes to complete a full build cycle.
+```bash
+./scripts/image-builder/build.sh -pull_secret_file ~/pull-secret.txt
```

Comment:
This step died with this error:

```
# Embedding kickstart in the installer image
sed: -e expression #1, char 36: unterminated `s' command
Execution time: 34 minutes
```

Reply:
Yes, this can happen when the pull secret contents are not compacted. I've added "jq -c" processing to normalize the contents before they're replaced.